processing/src/main/java/org/apache/carbondata/lcm/status/SegmentStatusManager.java

```diff
@@ -102,6 +91,60 @@ public long getTableStatusLastModifiedTime() throws IOException {
 
   /**
    * get valid segment for given table
+   *
+   * @return
+   * @throws IOException
+   */
+  public InvalidSegmentsInfo getInvalidSegments() throws IOException {
```

Comment:
This requires reading SegmentInfo twice, once for valid blocks and next for Invalid Blocks. Instead send a single class containting ValidAndInvalidBlocks.